### PR TITLE
111 - Remove duplicates from token list

### LINKED ISSUE #111: Duplicate tokens should not be allowed in the Token Manager

### PR DESCRIPTION
**Pull request by Preston-Harrison**
resolves #111

### CODE REVIEW

**Comment by rellfy** [comment]
Tests are failing

File: src/components/TokenManager/index.ts [hunk 1]
```diff
@@ -77,12 +77,34 @@ class TokenManager {
...
+
+      allTokens.forEach((token) => {
+        if (seen[tokenKey(token)]) {
+          noDuplicates.push(token);
+          seen[tokenKey(token)] = false;
+        }
+      });
+
+      returnTokens = noDuplicates;
+    }
```
Comment (rellfy) [inline src/components/TokenManager/index.ts:102]: This should be a private function.

File: src/components/TokenManager/index.ts [hunk 2]
```diff
@@ -77,12 +77,34 @@ class TokenManager {
   /**
    * @returns all tokens from all cached token lists
    */
-  public getLoadedTokens(network?: Network | number): TokenInfo[] {
+  public getLoadedTokens(network?: Network | number, removeDuplicates = false): TokenInfo[] {
```
Comment (rellfy) [inline src/components/TokenManager/index.ts:80]: It should default to true, I can't see an use case where you want duplicates.

**Review by rellfy — CHANGES_REQUESTED** [review]

File: src/components/TokenManager/index.ts (hunk 2 above)
Comment (Preston-Harrison) [reply src/components/TokenManager/index.ts:80]: Removing duplicates can involve alot of iterations over the entire token list, and for most cases when you are just getting a token by a symbol or address, duplicates are fine. Duplicates only need to be removed when the entire token list is needed at once

File: src/components/TokenManager/index.ts (hunk 1 above)
Comment (Preston-Harrison) [reply src/components/TokenManager/index.ts:102]: extracted to a public function as it is conceivable removing duplicate tokens might be needed outside of internal functions

**Review by rellfy — APPROVED** [review]

**Review by nabsta-eth — APPROVED** [review]

**Pull request merged**